actors/market/src/lib.rs

```diff
@@ -1065,25 +977,29 @@ where
     RT: Runtime<BS>,
 {
     let mut pieces = Vec::with_capacity(deal_ids.len());
+
+    let st: State = rt.state()?;
+
     for deal_id in deal_ids {
-        let deal = proposals
-            .get(*deal_id)
+        let deal = st
+            .deal_proposals_get(rt.store(), *deal_id)
```

Comment:
Please revert this to take the proposals as parameter. Even though we still pass the Runtime here for the syscall, this should look like a free function rather than stateful.

Reply:
Sure, Done  👍🏾